ethereum_history_api/circuits/lib/src/rlp/decode.nr

```diff
@@ -56,39 +51,39 @@ pub fn decode_to_rlp_header<N>(arr: [u8; N]) -> RlpHeader {
         RlpHeader { offset, length, data_type: LIST }
     } else { // >55-byte list
         let offset = wrapping_sub(1 + prefix, 0xf7) as u64;
-        let length = extract_payload_len(arr, wrapping_sub(prefix, 0xf7) as u64);
+        let length = extract_payload_len(data, wrapping_sub(prefix, 0xf7) as u64);
         assert(offset + length <= N, "Decoded length of long list exceeds input length");
 
         RlpHeader { offset, length, data_type: LIST }
     }
 }
 
-pub fn decode_string<N>(input: [u8; N]) -> (u64, u64) {
+pub fn decode_string<N>(input: Fragment<N, u8>) -> (u64, u64) {
```

Comment:
Let's name the argument: data. It would be consistent with other functions #data